sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_command_buffer.cpp

```diff
@@ -582,9 +583,138 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferAppendMembufferCopyRectExp(
                                  hCommandBuffer->Device));
 
   return enqueueCommandBufferMemCopyRectHelper(
-      hCommandBuffer, ZeHandleDst, ZeHandleSrc, srcOrigin, dstOrigin, region,
-      srcRowPitch, dstRowPitch, srcSlicePitch, dstSlicePitch,
-      numSyncPointsInWaitList, pSyncPointWaitList, pSyncPoint);
+      UR_COMMAND_MEM_BUFFER_COPY_RECT, hCommandBuffer, ZeHandleDst, ZeHandleSrc,
+      srcOrigin, dstOrigin, region, srcRowPitch, dstRowPitch, srcSlicePitch,
+      dstSlicePitch, numSyncPointsInWaitList, pSyncPointWaitList, pSyncPoint);
+}
+
+UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferAppendMembufferWriteExp(
+    ur_exp_command_buffer_handle_t hCommandBuffer,
+    ur_mem_handle_t hBuffer, ///< [in] handle of the buffer object
+    size_t offset,           ///< [in] offset in bytes in the buffer object
+    size_t size,             ///< [in] size in bytes of data being written
+    const void
+        *pSrc, ///< [in] pointer to host memory where data is to be written from
+    uint32_t numEventsInWaitList, ///< [in] size of the event wait list
+    const ur_exp_command_buffer_sync_point_t *pSyncPointWaitList,
+    ur_exp_command_buffer_sync_point_t *pSyncPoint) {
+
+  UR_ASSERT(hCommandBuffer, UR_RESULT_ERROR_INVALID_COMMAND_BUFFER_EXP);
+  UR_ASSERT(hBuffer, UR_RESULT_ERROR_INVALID_MEM_OBJECT);
+
+  ur_mem_handle_t_ *Buffer = ur_cast<ur_mem_handle_t_ *>(hBuffer);
```

Comment:
This cast seems unnecessary, is there a reason for it? Same for other similar uses further down.